src/sema/function_annotation.rs

```diff
@@ -0,0 +1,184 @@
+                                parsed_annotations.push(ConstructorAnnotation::Bump(expr));
+                            }
+                        }
+                    }
+                    continue;
+                }
+                "space" => {
+                    let ty = Type::Uint(64);
+                    let loc = note.loc;
+
+                    if let Ok(expr) = expression(
+                        &note.value,
+                        context,
+                        ns,
+                        symtable,
+                        &mut diagnostics,
+                        ResolveTo::Type(&ty),
+                    ) {
+                        if let Ok(expr) = expr.cast(&expr.loc(), &ty, true, ns, &mut diagnostics) {
+                            if let Some(prev) = &space {
+                                diagnostics.push(Diagnostic::error_with_note(
+                                    loc,
+                                    "duplicate @space annotation for constructor".into(),
+                                    *prev,
+                                    "previous @space".into(),
+                                ));
+                            } else {
+                                space = Some(loc);
+                                used_variable(ns, &expr, symtable);
+                                parsed_annotations.push(ConstructorAnnotation::Space(expr));
+                            }
+                        }
+                    }
+                    continue;
+                }
+                "payer" => {
+                    let ty = Type::Address(false);
+                    let loc = note.loc;
+
+                    if let Ok(expr) = expression(
+                        &note.value,
+                        context,
+                        ns,
+                        symtable,
+                        &mut diagnostics,
+                        ResolveTo::Type(&ty),
+                    ) {
+                        if let Ok(expr) = expr.cast(&expr.loc(), &ty, true, ns, &mut diagnostics) {
+                            if let Some(prev) = &payer {
+                                diagnostics.push(Diagnostic::error_with_note(
+                                    loc,
+                                    "duplicate @payer annotation for constructor".into(),
+                                    *prev,
+                                    "previous @payer".into(),
+                                ));
+                            } else {
+                                payer = Some(loc);
+                                used_variable(ns, &expr, symtable);
+                                parsed_annotations.push(ConstructorAnnotation::Payer(expr));
+                            }
+                        }
+                    }
+                    continue;
+                }
+                _ => (),
+            }
+        }
+
+        diagnostics.push(Diagnostic::error(
```

Comment:
This can go inside the match statement, in the `_ =>` branch.

Reply:
Rewritten. The code is slightly more readable now.